internal/helm/chart/dependency_manager_test.go

```diff
@@ -170,8 +199,125 @@ func TestDependencyManager_Build(t *testing.T) {
 			g.Expect(err).ToNot(HaveOccurred())
 
 			dm := NewDependencyManager(
-				WithRepositories(tt.repositories),
-				WithRepositoryCallback(tt.getChartRepositoryCallback),
+				WithRepositories(tt.downloaders),
+				WithDownloaderCallback(tt.getChartDownloaderCallback),
+			)
+			absBaseDir, err := filepath.Abs(tt.baseDir)
+			g.Expect(err).ToNot(HaveOccurred())
+			got, err := dm.Build(context.TODO(), LocalReference{WorkDir: absBaseDir, Path: tt.path}, chart)
+
+			if tt.wantErr != "" {
+				g.Expect(err).To(HaveOccurred())
+				g.Expect(err.Error()).To(ContainSubstring(tt.wantErr))
+				g.Expect(got).To(BeZero())
+				return
+			}
+
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(got).To(Equal(tt.want))
+			if tt.wantChartFunc != nil {
+				tt.wantChartFunc(g, chart)
+			}
+		})
+	}
+}
+
+func TestDependencyManager_OCIBuild(t *testing.T) {
```

Comment:
This test is identical to `TestDependencyManager_Build` with the only difference being the `mockRepo` function. Can we loop through the tests with the `mockRepo` (CleanableDownloader) as an argument?

Refer the tests for git managed and unmanaged transport where we pass an argument to run the same tests with different configurations https://github.com/fluxcd/source-controller/blob/d1f0d239e1b866f5039f9f75115445552e5a65cf/pkg/git/libgit2/checkout_test.go#L38 .